api/build.gradle.kts

```diff
@@ -35,3 +37,15 @@ dependencies {
 tasks.build {
   dependsOn("javadoc")
 }
+
+tasks.jar {
+  archiveBaseName.set(artifactName)
+}
+
+publishing {
+  publications {
+    withType<MavenPublication>().configureEach {
+      artifactId = artifactName
+    }
+  }
```

Comment:
Do we need to configure this one by one module, or we can achieve this in the main `build.gradle.kts`?

Reply:
I think it is achievable, you can implement this in `subprojects`, can you take a try?

Reply:
I'll take a look to see if this can be configured in the main `build.gradle.kts`.